framework/src/userobjects/SolutionUserObject.C

```diff
@@ -262,6 +262,9 @@ SolutionUserObject::readExodusII()
 
   // Build nodal/elemental variable lists, limit to variables listed in 'system_variables', if
   // provided
+  _nodal_variables.clear();
+  _elemental_variables.clear();
+  _scalar_variables.clear();
```

Comment:
Why is this change necessary? A comment might be good

Reply:
I have  derived class that is calling this function multiple times. So I need to clear these before doing the following `push_back`s. I've added a comment.

Reply:
thank you!